drivers/mtd/Kconfig

```diff
@@ -5,6 +5,48 @@
 # directory for more details.
 #
 
-config MODULE_MTD
+menuconfig MODULE_MTD
     bool "Memory Technology Device interface (MTD)"
     depends on TEST_KCONFIG
+
+if MODULE_MTD
+
+config MODULE_MTD_SPI_NOR
+    bool "MTD interface for SPI NOR Flash"
+    depends on HAS_PERIPH_SPI
+    select MODULE_PERIPH_SPI
+
+config MODULE_MTD_FLASHPAGE
+    bool "MTD interface for Flashpage"
+    depends on HAS_PERIPH_FLASHPAGE
+    depends on HAS_PERIPH_FLASHPAGE_PAGEWISE
+    select MODULE_PERIPH_FLASHPAGE
+    select MODULE_PERIPH_FLASHPAGE_PAGEWISE
+    help
+        Driver for internal flash devices implementing flashpage interface.
+
+config MODULE_MTD_NATIVE
+    bool "MTD native driver"
+    depends on NATIVE_OS_LINUX
+
+config MODULE_MTD_AT24CXXX
+    bool "MTD implementation for AT24CXXX"
+    depends on MODULE_AT24CXXX
+
+config MODULE_MTD_AT25XXX
+    bool "MTD implementation for AT25XXX"
+    depends on MODULE_AT25XXX
+
+config MODULE_MTD_MCI
+    bool "MTD interface for LPC23XX MCI"
+    depends on MODULE_MCI
+
+config MODULE_MTD_SDCARD
+    bool "MTD interface for SPI SD-Card"
+    depends on MODULE_SDCARD_SPI
```

Comment:
What do you think about moving all these into an "Interfaces" menu and also moving `MODULE_MTD_WRITE_PAGE` inside the `if`, so we drop the `depends on MODULE_MTD`?